src/plugins/intel_gpu/src/runtime/memory_pool.cpp

```diff
@@ -240,6 +240,36 @@ memory::ptr memory_pool::get_memory(const layout& layout,
     }
 }
 
+memory_ptr memory_pool::get_intermediate_memory(const layout& layout,
```

Comment:
1) Why do we need to reuse intermediate memory with a separate handle? 
2) Can it handle the case where a primitive has multiple intermediates memories?

Reply:
1. A has not been recycled so far, and I wanted to prevent unnecessary side effects by adding one struct object and one function.
2. I will check this out.

Reply:
I've checked that this function and a separate handle are not needed. It is enough to use `memory_pool::get_memory()` function. I will remove them.

Reply:
Own id is added to mem deps. And `memory_pool::get_memory()` will be used to intermediate memories. Anyway, there is no multiple intermediate memory issue. It checked with DetectionOutput unit test.